File: test/rsolv/ast/fixtures/mock_parser.rb
#!/usr/bin/env ruby
# Mock Ruby parser for testing Port supervision.

require 'json'

STDOUT.sync = true
STDIN.sync = true

while line = STDIN.gets
  begin
    request = JSON.parse(line.strip)
    
    response = case request['command']
    when 'HEALTH_CHECK'
      { 'status' => 'healthy', 'id' => request['id'] }
    when 'parse'
      # Simulate AST parsing
      {
        'id' => request['id'],
        'result' => {
          'ast' => { 'type' => 'Program', 'body' => [] },
          'language' => 'ruby',
          'parser_version' => '1.0.0'
        }
      }
    else
      {
        'id' => request['id'],
        'result' => { 'echo' => request['command'] || 'unknown' }
      }
    end
    
    puts JSON.generate(response)
  rescue => e
    error_response = {
      'id' => request ? request['id'] : nil,
      'error' => e.message
    }
    puts JSON.generate(error_response)
  end
end

File: test/rsolv/ast/fixtures/mock_parser.py
#!/usr/bin/env python3
"""Mock Python parser for testing Port supervision."""

import sys
import json
import time

def main():
    while True:
        try:
            line = sys.stdin.readline()
            if not line:
                break
                
            request = json.loads(line.strip())
            
            # Handle different test commands
            if request.get("command") == "CRASH_NOW":
                sys.exit(1)
            elif request.get("command") == "HEALTH_CHECK":
                response = {"status": "healthy", "id": request.get("id")}
            elif request.get("command") == "parse":
                # Simulate AST parsing
                response = {
                    "id": request.get("id"),
                    "result": {
                        "ast": {"type": "Program", "body": []},
                        "language": "python",
                        "parser_version": "1.0.0"
                    }
                }
            else:
                response = {
                    "id": request.get("id"),
                    "result": {"echo": request.get("command", "unknown")}
                }
            
            # Send response
            print(json.dumps(response))
            sys.stdout.flush()
            
        except Exception as e:
            error_response = {
                "id": request.get("id") if 'request' in locals() else None,
                "error": str(e)
            }
            print(json.dumps(error_response))
            sys.stdout.flush()

if __name__ == "__main__":
    main()